### ISSUE TITLE
Bug: Description is duplicated when window loses focus while editing

### ISSUE THREAD

**Issue by Johennes** [mattermost/focalboard#309]
**Summary:**

When editing the description on a new card and then switching to another application before leaving the text field, the description is duplicated.

**Steps to reproduce the behavior:**

1. Create a new card
2. Enter text into the description field
3. Without leaving the description field, switch to another app
4. Return to Focalboard
5. Notice how the description field has been duplicated

**Expected behavior:**

The description field should either stay in editing mode or it should leave editing but there should be only one description when returning.

**Screenshots:**

Before switching to another app
![Screenshot 2021-04-21 at 10 43 27](https://user-images.githubusercontent.com/1137962/115524865-d9ea5180-a28e-11eb-8815-fe021fec2562.png)

After returning to Focalboard
![Screenshot 2021-04-21 at 10 43 36](https://user-images.githubusercontent.com/1137962/115524870-da82e800-a28e-11eb-8505-17b8a47cde07.png)

**Platform:**

macOS 11.2.1
Focalboard v0.6.0

**Comment by chenilim** [comment]
Oh wow, great bug @Johennes! Confirm this repros on the new v0.6.5 release too (Mac App Store app) as well as when using Safari against the hosted Personal Serve web app. However, it does not repro with Chrome. Appears to be a timing issue with Safari / WKWebView.

**Comment by welteki** [comment]
I can confirm this issue is reproducible on v0.6.5 Mac App Store app and using Safari against the server.

**Comment by Johennes** [comment]
It seems that the duplicate description stems from the [`onBlur`](https://github.com/mattermost/focalboard/blob/main/webapp/src/components/cardDetail/cardDetailContents.tsx#L83) handler in `CardDetailContents` triggering twice. That in turn results from calling [`instance.getInputField()?.blur()`](https://github.com/mattermost/focalboard/blob/main/webapp/src/components/markdownEditor.tsx#L120) inside the `blur` handler in `MarkdownEditor`.

I have tried commenting out `instance.getInputField()?.blur()` and that fixed the issue in Safari / the mac app. It continued working correctly in Firefox regardless of whether that line was commented or not.

Based on these findings, I'd suggest to simply remove this line. However, since blurring the input field _inside_ the blur handler looks rather suspicious, I'm not sure if maybe this was added intentionally to work around some other bug occurring in different circumstances?

If we cannot remove this, alternatively we could prevent the second invocation of `onBlur` by leveraging the `active` state variable.
